subsys/net/openthread/Kconfig.defconfig

```diff
@@ -10,6 +10,10 @@
 
 if NET_L2_OPENTHREAD
 
+choice OPENTHREAD_IMPLEMENTATION
+	default OPENTHREAD_LIBRARY
```

Comment:
Building from libraries fails for `nrf21540dk_nrf52840`.

Reply:
What about conditioning the selection of the libraries for the supported SoCs only?